modules/electrophysiology_browser/php/sessions.class.inc

```diff
@@ -47,11 +47,43 @@ class Sessions extends \NDB_Page
      */
     function _hasAccess(\User $user) : bool
     {
-        return ($user->hasPermission('electrophysiology_browser_view_allsites')
+        return (($user->hasPermission('electrophysiology_browser_view_allsites')
                 || ($user->hasCenter($this->timepoint->getCenterID())
-                && $user->hasPermission('electrophysiology_browser_view_site')
-            )
-        );
+                && $user->hasPermission('electrophysiology_browser_view_site'))
+            ) && $user->hasProject($this->candidate->getProjectID()));
+    }
+
+    /**
+     * Load the required variables in order to check that the  user
+     * has access to the session.
+     *
+     * @param \User                  $user    The user to load the resources for
+     * @param ServerRequestInterface $request The PSR15 Request being handled
+     *
+     * @throws \DomainException If the session id is non-numerical
+     * @throws \LorisException  If the session is not found
+     *
+     * @return void
+     */
+    public function loadResources(
+        \User $user, ServerRequestInterface $request
+    ) : void {
+
+        $path    = $request->getUri()->getPath();
+        $matches = [];
+
+        if (preg_match('#/sessions/(\d+)#', $path, $matches) !== 1) {
+            throw new \DomainException("Invalid session");
```

Comment:
This is going to cause a 500 error when the path is invalid, which is incorrect, it should still be a 404 error. It would be better to do something like:

Some solutions I can think of off the top of my head:
1. Set a class variable (something like "invalidSession = true") and return if that happens. Return "true" for hasAccess if invalid so that handle still gets called. Return the 404 from handle
2. change loadResources to return an error code instead of void, modify the module class to check it and return a 404 if the request was invalid (this assumes that the only reason an error would be returned is an invalid request, which is probably true.)

@ridz1208 any thoughts on which you prefer?

Reply:
I can take a crack at both those options and let you know which works better or which is simpler. Thanks!

Reply:
Option 3 is probably a better idea: change the exception type (again) to a NotFound exception so that the try/catch around the loadResources handles the exception to return a 404.